libs/contrib/Text/Parser.idr

```diff
@@ -4,10 +4,20 @@ import Data.Bool.Extra
 
 import public Text.Parser.Core
 import public Text.Quantity
+import public Text.Token
 
 %access export
 %default total
 
+||| Parse a terminal based on a kind of token.
+match : (Eq k, TokenKind k) =>
```

Comment:
It might be better to add the `Eq` interface constraint to the definition for `TokenKind` itself. This way you are forced to define an `Eq` implementation during the definition of the `TokenKind` implementation.

Reply:
I considered this, but decided against it because it seems like a valid option to use `Token` and `TokenKind` without using `match`.